### DISCUSSION TITLE
How to jump to previous cursor ?

### DISCUSSION THREAD

**Discussion by dytra** [VSCodeVim/Vim#9052] [Q&A]
I'm used using neovim extension in vscode. but because it is unstable, I wanna try this vim extension.
But I have an issue.

I often use the keybind `gd` to navigate to the definition of a function in neovim. then press `CTRL+O` to jump to previous cursor.
I can use the `gd` keybind in vim, but when I press the `CTRL+O`, it makes a new line, which not I want.
How can I achieve this ?

**Comment by Tusenka** [comment]
did you find the solution?

**Reply by dytra** [reply]
I don't have solution for this, I just use neovim vscode extension instead

**Comment by xiays146** [comment]
when using ubuntu, there is a vscode key for juming to previous cursor, like 'ctrl' + 'alt' + '-' 
it is not a vim key, but solve your problem.

**Comment by bzy-debug** [comment]
I personally have two solutions:

1. use vscode's builtin navigate command:

    Add following to vim.normalModeKeyBindingsNonRecursive:
    
    ```json
    {
      "before": ["<C-o>"],
      "commands": ["workbench.action.navigateBack"]
    },
    {
      "before": ["<C-i>"],
      "commands": ["workbench.action.navigateForward"]
    },
    ```
    
    However, this setting does not play well with `<C-d>`. For example, if `<C-d>` three times after `gd`, you need to to tap `<C-o>` four times (three times for the `<C-d>`, one time for `gd`) to go back.

2. try out my extension: https://marketplace.visualstudio.com/items?itemName=bzy-debug.vim-jumplist . It aims to solve the `<C-d>` problem of solution 1.

**Comment by h3yng** [comment]
`gd`, `CTRL+o`, and `CTRL+i`  works fine, you might have  conflicting keymap.
